apis/keda/v1alpha1/cloudevent_types.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2021 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CloudEvent defines how a trigger can authenticate globally
+// +genclient
+// +kubebuilder:resource:path=cloudevents,scope=Namespaced
+// +kubebuilder:subresource:status
+type CloudEvent struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec CloudEventSpec `json:"spec"`
+}
+
+// +kubebuilder:object:root=true
+
+// CloudEventList is a list of CloudEvent resources
+type CloudEventList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []CloudEvent `json:"items"`
+}
+
+// CloudEvent
+type CloudEventSpec struct {
+	// +optional
+	ClusterName string `json:"clusterName,omitempty"`
```

Comment:
I think that this is for enriching the event information, is it correct? If yes, I think that we can be more efficient adding en environment variable that users can set for setting the value at KEDA level. I mean, the clusterName will be always the same because all the events are emitted by the same cluster.
As default value, we could do introspection of the service account token and use the token issuer. It'll use the cluster issuer (which should be unique) as default

Reply:
Cluster name can mean different things to different people. For a single team  it can be "prod" while for others it can be "prod-eu" for example.

Design proposal defined what you proposed: https://docs.google.com/document/d/14Z04AWiMBkSX88uiaVS-kg-8QjjNcDhVy6-u5FMUk_w/edit#heading=h.z98yu2fwr49j but I actually like giving options and doing both might be ideal where cluster name in CRD wins over KEDA default.

Thoughts?

If so, no need to change here IMO and we can open separate issue to avoid creating monolithical change

Reply:
Yeah, I like that option. The precedence could be something like CRD name > KEDA environment > token issuer. This allows admins to set their name and other users can override the naming on their own CR.